pkg/util/util.go

```diff
@@ -72,15 +73,32 @@ func IsVFIOVMI(vmi *v1.VirtualMachineInstance) bool {
 	return false
 }
 
-func NeedVirtioNetDevice(vmi *v1.VirtualMachineInstance, useEmulation bool) bool {
+func NeedVirtioNetDevice(vmi *v1.VirtualMachineInstance, allowEmulation bool) bool {
 	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if !useEmulation && (iface.Model == "" || iface.Model == "virtio") {
+		if !allowEmulation && (iface.Model == "" || iface.Model == "virtio") {
 			return true
 		}
```

Comment:
ugh... so this isn't 100% accurate i guess.  More accurately, we'd need to know if emulation is actually in use for the VMI or not.

This logic wasn't right to begin with in mixed mode setups where some nodes use emulation and others use kvm... I've never heard of such a setup, but I guess that's what this PR accounting for?

Reply:
> but I guess that's what this PR accounting for

Not really actually, this PR simply tries to have the code use a more accurate terminology, to avoid misinterpretations of the useEmulation semantics. This use here is a good example of a misinterpretation, with mixed mode setup or not :-)

But I want to better understand the problematic case.
If `allowEmulation == true`, then `NeedVirtioNetDevice()` returns `false`. Which means two things:
1. Device ownership isn't claimed by virt-handler for a non-root VMI. I'm pretty sure this one needs to be fixed.
2. virt-controller's template rendering logic won't set the vhost-net resource requests for the scheduler. This means that if a VMI does land on a node with vhost-net device, then it won't consume any of the node's `devices.kubevirt.io/vhost-net` integer resources. Do you think this imposes a problem?